lib/puppet/type/sensu_asset.rb

```diff
@@ -76,7 +76,7 @@
   newparam(:resource_name, :namevar => true) do
     desc "The name of the asset."
     validate do |value|
-      unless value =~ /^[\w\.\-\/]+$/
+      unless value =~ %r{^[a-z0-9\/\_\.\-\:]+$}
```

Comment:
why does asset use a regex instead of `PuppetX::Sensu::Type.name_regex` ?

Reply:
Because this is how Sensu Go does it. The name regex used by Asset type is not the one shared by other resources. The Asset does not support uppercase letters and also allows slashes when others don't share that pattern. I pulled regex from Sensu Go source.

Reply:
Each regex should be documented in a way that is easy to read. By using symbols and commas it makes it hard to understand. Below is an example comment for a regex.

Match strings that begin with and contain only letters, numbers, underscores, periods and colons. The string may also be empty.

Reply:
Updated comments per suggestion.